### PR TITLE
Create codeql.yml

### PR DESCRIPTION
**Pull request by Birdbh**
Added QL Analysis workflow as a part of the CI/CD pipeline based on professor resources

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      security-events: write
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ python ]
```
Comment (JamesShadowfyre) [inline .github/workflows/codeql.yml:43]:
```suggestion
        language: [ 'python' ]
```

**Review by JamesShadowfyre — CHANGES_REQUESTED** [review]
Fixing code

**Review by Jcp149 — APPROVED** [review]
Looks good to me assuming it works when we actually have code to run!

**Review by JamesShadowfyre — APPROVED** [review]
Passes all tests when run now. Looks good to merge!

**Pull request merged**